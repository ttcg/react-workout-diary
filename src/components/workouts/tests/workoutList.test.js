import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WorkoutList from '../workoutList';

describe('WorkoutListTest', () => {
    it('renders without crashing', () => {
        const showEdit = jest.fn(),
            handleDelete = jest.fn(),
            items = [
                {
                    "id": '38046275-fe56-414e-9c51-75e2fc84438e',
                    "workoutType": "Running",
                    "date": "2019-01-02",
                    "calories": 340
                },
                {
                    "id": '28046275-fe56-414e-9c51-75e2fc84438f',
                    "workoutType": "Running",
                    "date": "2019-01-02",
                    "calories": 500
                }
            ]

        const component = shallow(<WorkoutList items={items} handleDelete={handleDelete} showEdit={showEdit} />);

        const tree = toJson(component);
        expect(tree).toMatchSnapshot();

        expect(component.find('WorkoutRow')).toHaveLength(items.length);
    });
});