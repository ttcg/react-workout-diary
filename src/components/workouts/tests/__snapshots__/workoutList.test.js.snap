// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutListTest renders without crashing 1`] = `
<Table
  responsiveTag="div"
  striped={true}
  tag="table"
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Date
      </th>
      <th>
        Type
      </th>
      <th>
        Calories
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <WorkoutRow
      handleDelete={[MockFunction]}
      i={0}
      item={
        Object {
          "calories": 340,
          "date": "2019-01-02",
          "id": "38046275-fe56-414e-9c51-75e2fc84438e",
          "workoutType": "Running",
        }
      }
      key="0"
      showEdit={[MockFunction]}
    />
    <WorkoutRow
      handleDelete={[MockFunction]}
      i={1}
      item={
        Object {
          "calories": 500,
          "date": "2019-01-02",
          "id": "28046275-fe56-414e-9c51-75e2fc84438f",
          "workoutType": "Running",
        }
      }
      key="1"
      showEdit={[MockFunction]}
    />
  </tbody>
</Table>
`;
